dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,130 @@
+# RFC: Names for H3 Concepts, Types, and Functions
+
+- **Authors**: AJ Friend
+- **Date**: 2020-02-02
+- **Status**: Draft
+
+## Motivation
+
+Concepts like `hexagon`, `cell`, and `index` are currently used ambiguously in the H3 codebase and in documentation.
+This can cause confusion, for example, when a function might only work on strict H3 *hexagons*, but fails when encountering H3 *pentagons*.
+
+Reaching a consensus on the precise, technical language used when discussing H3 concepts will clarify future library discussions and improve end-user documentation.
+
+## Proposals
+
+## Concepts
+
+The following technical terms should be used in the documentation, the H3 codebase, and precise technical discussions of the library.
+
+- `H3Index`:
+    - an unsigned 64-bit integer representing **any** H3 object (hexagon, pentagon, directed edge, ...)
+    - often represented as a 15-character (or 16-character) hexadecimal string, like `'8928308280fffff'`
+    - the full term "H3 index" should be used to avoid confusion with other common uses of "index"; when a "traditional" index is needed, prefer using "number", "pos", or another term to avoid confusion
+- `Cell`:
+    - an H3 index of `mode 1` (hexagon or pentagon)
+    - for functions that can handle either hexagons or pentagons, the more general term "cell" should be used whenever possible
+- hexagon:
+    - an H3 cell which is a **topological** hexagon
+    - todo: advise use of the term "strict" for functions that can only handle hexagons?
+- pentagon:
+    - an H3 cell which is a **topological** pentagon
+- `DirectedEdge`
+    - an H3 index of `mode 2`
+    - representing a traversal from an origin cell to an adjacent destination cell
+- grid:
+    - the graph with nodes corresponding to H3 cells, and edges given by pairs of adjacent cells
+    - for example, `gridDistance` is the minimal number of edges in a graph path connecting two cells
```

Comment:
@sahrk Would you use "grid" or "graph" to describe the topological connections between adjacent cells? Or is there another term that's better?

Reply:
Grid! It's right there in the name. Got it, got it. Cool. Moving along... :man_facepalming:

Reply:
> @sahrk Would you use "grid" or "graph" to describe the topological connections between adjacent cells? Or is there another term that's better?

I have always seen and used "grid".

Reply:
Cool. Case closed!